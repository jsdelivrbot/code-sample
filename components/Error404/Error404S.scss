@import "guide/includes";

.Error404 {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-top: 20%;

  h3 {
    text-align: center;
  }

  .FunLoader {
    margin-top: 35px;
  }
}
