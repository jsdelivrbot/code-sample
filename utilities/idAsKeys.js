import keyBy from 'lodash/keyBy'
import { array, checkPropTypes } from 'prop-types'

const propTypes = {
  items: array,
}

const idAsKeys = function ({ items }) {
  checkPropTypes(propTypes, arguments[0], 'prop', 'idAsKeys')

  return keyBy(items, (item) => {
    return item.data.id
  })
}

export default idAsKeys
