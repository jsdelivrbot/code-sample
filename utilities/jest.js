export const describe = global.describe

export const it = global.it

export const expect = global.expect

export const beforeEach = global.beforeEach

export const afterEach = global.afterEach
