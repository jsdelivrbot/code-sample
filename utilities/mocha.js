export const describe = window.describe

export const it = window.it

export default window.mocha
