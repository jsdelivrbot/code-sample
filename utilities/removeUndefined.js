import merge from 'lodash/merge'
import forIn from 'lodash/forIn'
import isObject from 'lodash/isObject'

// TODO: use destructor
function removeUndefined (o) {
  if (isObject(o) && !Array.isArray(o)) {
    let obj = merge({}, o)

    forIn(obj, function (val, key) {
      if (val === null || val === undefined) {
        delete obj[key]
      } else {
        obj[key] = removeUndefined(val)
      }
    })

    return obj
  }

  return o
}

export default removeUndefined
