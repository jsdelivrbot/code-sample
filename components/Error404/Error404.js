import './Error404S.scss'
import React from 'react'
import FunLoader from 'components/FunLoader/FunLoader'

const Error404 = () => {
  return (
    <div className='Error404'>
      <h3>Oops, the page you're looking for doesn't exist.</h3>

      <FunLoader />
    </div>
  )
}

Error404.propTypes = {
}

export default Error404
