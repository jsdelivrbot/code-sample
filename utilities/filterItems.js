import filter from 'lodash/filter'
import isMatch from 'lodash/isMatch'
import removeUndefined from 'utilities/removeUndefined'
import { object, checkPropTypes } from 'prop-types'

const propTypes = {
  items: object,
  filters: object,
}

const filterItems = function ({ items, filters }) {
  checkPropTypes(propTypes, arguments[0], 'prop', 'filterItems')

  const removedUndefinedFilters = removeUndefined(filters)

  return filter(items, ({ data }) => {
    return isMatch(data, removedUndefinedFilters)
  })
}

export default filterItems
