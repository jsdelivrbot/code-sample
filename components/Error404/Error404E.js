import React from 'react'
import Error404 from './Error404'

const Error404E = function () {
  return (
    <div className='Error404E'>
      <h1 className='exampleHeader'>Default</h1>

      <Error404 />
    </div>
  )
}

export default Error404E
