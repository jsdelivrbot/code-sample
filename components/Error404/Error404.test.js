import Error404 from './Error404'
import React from 'react'
import renderer from 'react-test-renderer'
import { describe, it, expect } from 'utilities/jest'

describe('Test Error404', () => {
  it('should be able to render the component', () => {
    const component = renderer.create(
      <Error404 />
    )

    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
