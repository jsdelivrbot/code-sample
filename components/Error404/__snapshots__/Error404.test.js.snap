// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Error404 should be able to render the component 1`] = `
<div
  className="Error404"
>
  <h3>
    Oops, the page you're looking for doesn't exist.
  </h3>
  <div
    className="FunLoader"
  >
    <span
      className="fa fa-grav"
    />
  </div>
</div>
`;
